collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -800,13 +804,18 @@ protected void onActivityResult(int requestCode, int resultCode, final Intent in
             case RequestCodes.EX_STRING_CAPTURE:
             case RequestCodes.EX_INT_CAPTURE:
             case RequestCodes.EX_DECIMAL_CAPTURE:
-                String key = "value";
-                boolean exists = intent.getExtras().containsKey(key);
-                if (exists) {
-                    Object externalValue = intent.getExtras().get(key);
-                    if (getCurrentViewIfODKView() != null) {
-                        setWidgetData(externalValue);
-                    }
+                Object externalValue = externalAppIntentProvider.getValueFromIntent(intent);
+                if (getCurrentViewIfODKView() != null) {
+                    setWidgetData(externalValue);
+                }
+                break;
+            case RequestCodes.EX_ARBITRARY_FILE_CHOOSER:
+                if (intent.getClipData() != null
```

Comment:
You'll probably want to pull this out somewhere once you have more external file widgets but this looks fine for now!